src/App.tsx

```diff
@@ -1,8 +1,92 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { getGoods } from './components/GOODS_FROM_SERVER_API';
+import { Good } from './types';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface AppStateInterface {
+  goodsList: Good[];
+}
 
-export default App;
+export default class App extends Component<{}, AppStateInterface> {
+  state = {
+    goodsList: [],
+  };
+
+  loadGoods = () => {
+    getGoods()
+      .then((goodsList) => {
+        this.setState({ goodsList });
+      });
+  }
+
+  firstFiveSorted = () => {
+    getGoods()
+    .then((goodsList) => {
+      this.setState({
+        goodsList: goodsList.sort((a, b) => a.name.localeCompare(b.name)).splice(0, 5),
+      });
+    })
+  }
+
+  showedRed = () => {
+    getGoods()
+    .then((goodsList) => {
+      this.setState({
+        goodsList: goodsList.filter(good => good.color === 'red'),
+      });
+    })
+  }
+
+  showedAllGoods = () => {
+    getGoods()
+      .then((goodsList) => {
+        this.setState({ goodsList });
+      });
+  }
+
+
+  render() {
+    const { goodsList } = this.state;
+
+    if (goodsList.length === 0) {
+      return (
+        <>
+          <p>
+            Load your Goods
+          </p>
+          <button
+            type="button"
+            onClick={this.loadGoods}>
+            Load
+          </button>
+        </>
+      )
+    }
```

Comment:
better to use ternary inside div.App and always render <h1>Goods</h1>

also, you do not really need separate start button just render or not GoodsList and always render 3 buttons